include/triton/core/tritonserver.h

```diff
@@ -1634,6 +1634,18 @@ TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_ServerLoadModel(
 TRITONSERVER_DECLSPEC TRITONSERVER_Error* TRITONSERVER_ServerUnloadModel(
     TRITONSERVER_Server* server, const char* model_name);
 
+/// Unload the requested model, and also unload the models that only be used
+/// by the requested model. Unloading a model that is not loaded
+/// on server has no affect and success code will be returned.
+/// The function does not return until the model is unloaded or fails to unload.
+/// Returned error indicates if model unloaded successfully or not.
```

Comment:
What happens if only some of the models fail to unload. Do you still unload the others? Is the returned error just the error from one of the failed unloads?

Reply:
Actually there is no failure case for unloading a model. Even in the case of there is inflight requests, this function doesn't really wait for the unload to complete, it let model manager to drop its smart pointer to the model and return success.

Reply:
We should fix the comment then, and in the ServerUnloadModel function as well.